src/extraction/FStar.Extraction.ML.Modul.fst

```diff
@@ -751,13 +751,22 @@ let mark_sigelt_erased (se:sigelt) (g:uenv) : uenv =
   List.fold_right (fun lid g -> extend_erased_fv g (S.lid_as_fv lid None))
                   (U.lids_of_sigelt se) g
 
+// If the definition has an [@@extract_as impl] attribute,
+// replace the lbdef with the specified impl:
+let fixup_sigelt_extract_as se =
+  match se.sigel, find_map se.sigattrs N.is_extract_as_attr with
+  | Sig_let {lids; lbs=(_, [lb])}, Some impl ->
+    {se with sigel = Sig_let {lids; lbs=(true, [{lb with lbdef = impl}])}}
```

Comment:
The `true` is interesting here, so we always mark it recursive? Can it really be so since the term that is the argument to `extract_as` cannot mention the name of this sigelt (without pack_fv trickery)?

Reply:
In any case it's not a big deal, but maybe worth a comment in the attribute's doc and/or here.

Reply:
As mentioned offline, the implementation can indeed be recursive.  This is for example the case with recursive Pulse functions.

I have added a comment to both the attribute and the fixup function here.